lib/prng/fastrandombytes.cpp

```diff
@@ -4,44 +4,29 @@
  * Public Domain
  */
 
-#include "nfl/prng/crypto_stream_salsa20.h"
-#include "nfl/prng/randombytes.h"
 #include <inttypes.h>
 #include <iostream>
+#include "nfl/prng/crypto_stream_salsa20.h"
+#include "nfl/prng/randombytes.h"
+
+namespace nfl {
 
 static int init = 0;
-static unsigned char key[crypto_stream_salsa20_KEYBYTES];
-static unsigned char nonce[crypto_stream_salsa20_NONCEBYTES] = {0};
+static unsigned char key[32];
+static unsigned char nonce[8] = {0};
 
-void fastrandombytes(unsigned char *r, unsigned long long rlen)
-{
-  unsigned long long n=0;
+void fastrandombytes(unsigned char *r, unsigned long long rlen) {
+  unsigned long long n = 0;
   int i;
-  if(!init)
-  {
-    randombytes(key, crypto_stream_salsa20_KEYBYTES);
+  if (!init) {
+    randombytes(key, 32);
     init = 1;
   }
-  //crypto_stream(r,rlen,nonce,key);
-  crypto_stream_salsa20(r,rlen,nonce,key);
+  nfl_crypto_stream_salsa20_amd64_xmm6(r, rlen, nonce, key);
```

Comment:
It seems our code for randomness generation only works on amd64 architectures, doesn't it? @carlosaguilarmelchor